sympy/combinatorics/tests/test_rewriting.py

```diff
@@ -21,3 +25,16 @@ def test_rewriting():
     # but also the system should actually be confluent
     assert R._check_confluence()
     assert G.reduce(b*a**-1*b**-1*a**3*b**4*a**-1*b**-15) == a**-1*b**-1
+    # check for automaton reduction
+    assert R.reduce_using_automaton(b*a**-1*b**-1*a**3*b**4*a**-1*b**-15) == a**-1*b**-1
+
+    G = FpGroup(F, [a**2, b**3, (a*b)**4])
+    R = G._rewriting_system
+    assert G.reduce(a**2*b**-2*a**2*b) == b**-1
+    assert R.reduce_using_automaton(a**2*b**-2*a**2*b) == b**-1
+    assert G.reduce(a**3*b**-2*a**2*b) == a**-1*b**-1
+    assert R.reduce_using_automaton(a**3*b**-2*a**2*b) == a**-1*b**-1
+    # Check after adding a rule
+    R.add_rule(a**2, b)
+    assert R.reduce_using_automaton(a**2*b**-2*a**2*b) == b**-1
```

Comment:
Just noticed this: is this correct? With the rule you added, we should have `R.reduce_using_automaton(a**2*b**-2*a**2*b) == b`, even if `b**-1` is equivalent to `b` under the rule, `b` is smaller in the ordering that the rewriting system enforces on generators. What does `reduce` return in this case? Does `G.equals(b, b**-1)` work? You could also try running `make_confluent` after adding the rule and seeing if that makes a difference.

Reply:
> What does reduce return in this case? 

`reduce` returns `b**-1` in Python3 and `b` in Python 2.

Reply:
Even after `make_confluent`?